pkg/apis/openebs.io/v1alpha1/cstorpool_cluster.go

```diff
@@ -24,72 +24,72 @@ import (
+type CStorPoolCluster struct {
 	metav1.TypeMeta   `json:",inline"`
 	metav1.ObjectMeta `json:"metadata,omitempty"`
-	Spec              StoragePoolClaimSpec   `json:"spec"`
-	Status            StoragePoolClaimStatus `json:"status"`
+	Spec              CStorPoolClusterSpec   `json:"spec"`
+	Status            CStorPoolClusterStatus `json:"status"`
 }
 
-// StoragePoolClaimSpec is the spec for a StoragePoolClaimSpec resource
-type StoragePoolClaimSpec struct {
+// CStorPoolClusterSpec is the spec for a CStorPoolClusterSpec resource
+type CStorPoolClusterSpec struct {
 	Name     string                     `json:"name"`
 	Type     string                     `json:"type"`
 	MaxPools *int                       `json:"maxPools"`
 	MinPools int                        `json:"minPools"`
-	Nodes    []StoragePoolClaimNodeSpec `json:"nodes"`
+	Nodes    []CStorPoolClusterNodeSpec `json:"nodes"`
 	PoolSpec CStorPoolAttr              `json:"poolSpec"`
 }
 
-// StoragePoolClaimNodeSpec is the spec for node where pool should be created.
-type StoragePoolClaimNodeSpec struct {
+// CStorPoolClusterNodeSpec is the spec for node where pool should be created.
+type CStorPoolClusterNodeSpec struct {
 	// Name is the name of the node.
 	Name string `json:"name"`
 	// PoolSpec is the pool related specification that is used to provision cstor pool on the node.
 	PoolSpec CStorPoolAttr `json:"poolSpec"`
 	// DiskGroups contains the list of disk groups that should be used for pool provisioning on that node.
-	DiskGroups []StoragePoolClaimDiskGroups `json:"groups"`
+	DiskGroups []CStorPoolClusterDiskGroups `json:"groups"`
 }
 
-// StoragePoolClaimDiskGroups contains details of a disk group.
-type StoragePoolClaimDiskGroups struct {
+// CStorPoolClusterDiskGroups contains details of a disk group.
+type CStorPoolClusterDiskGroups struct {
 	// Name is the name of the disk group.
 	Name string `json:"name"`
 	// Disks is the disks present in the group.
-	Disks []StoragePoolClaimDisk `json:"disks"`
+	Disks []CStorPoolClusterDisk `json:"disks"`
 }
 
-// StoragePoolClaimDisk contains the name details of a disk CR.
-type StoragePoolClaimDisk struct {
+// CStorPoolClusterDisk contains the name details of a disk CR.
+type CStorPoolClusterDisk struct {
 	// Name is the name of the disk CR.
 	Name string `json:"name"`
 	// ID is a unique id associated to this disk either by user or system.
 	// This ID does not change even if the disk is replaced so one can think it as a slot for this disk.
 	ID string `json:"id"`
 }
 
-// StoragePoolClaimStatus is for handling status of pool.
-type StoragePoolClaimStatus struct {
+// CStorPoolClusterStatus is for handling status of pool.
+type CStorPoolClusterStatus struct {
 	Phase string `json:"phase"`
 }
 
-// CStorPoolAttr is to describe zpool related attributes.
-type CStorPoolAttr struct {
+// CStorPoolClusterSpecAttr is to describe zpool related attributes.
+type CStorPoolClusterSpecAttr struct {
```

Comment:
Can we use string constant as datatype for PoolType?

Reply:
Will fix it.

Reply:
fixed.